### DISCUSSION TITLE
Unable to install latest SB version: Error: ENAMETOOLONG: name too long, unlink...

### DISCUSSION THREAD

**Discussion by rogix** [storybookjs/storybook#18441] [General] [answered]
When I try to install Storybook versions 6.5.6 or any version starting from 6.5 I get the error printed bellow. If I install 6.4.22 it works fine. 

I am using linux mint with the `/home` folder encrypted with encryptfs (not sure if this has anything to do with the issue, but I think it's a important information). 

Has anyone here faced a similar problem? Is this a SB issue? Any idea? Thanks!

```
Arguments: 
  /home/username/.nvm/versions/node/v14.17.5/bin/node /home/username/.yarn/bin/yarn.js add -D --ignore-workspace-root-check @storybook/react@^6.5.6 @storybook/addon-links@^6.5.6 @storybook/addon-essentials@^6.5.6 @storybook/addon-actions@^6.5.6 @storybook/addon-interactions@^6.5.6 @storybook/testing-library@^0.0.11 @storybook/builder-webpack4@^6.5.6 @storybook/manager-webpack4@^6.5.6 @babel/core@^7.18.2 babel-loader@^8.2.5

PATH: 
  /home/username/code/personal/open-source/some/project/node_modules/.bin:/home/username/code/personal/open-source/some/node_modules/.bin:/home/username/code/personal/open-source/node_modules/.bin:/home/username/code/personal/node_modules/.bin:/home/username/code/node_modules/.bin:/home/username/node_modules/.bin:/home/node_modules/.bin:/node_modules/.bin:/home/username/.nvm/versions/node/v14.17.5/lib/node_modules/npm/node_modules/@npmcli/run-script/lib/node-gyp-bin:/home/username/.npm/_npx/6fe9a9991b157df1/node_modules/.bin:/home/username/.yarn/bin:/home/username/.config/yarn/global/node_modules/.bin:/home/username/.yarn/bin:/home/username/.config/yarn/global/node_modules/.bin:/home/username/.zinit/polaris/bin:/home/username/.nvm/versions/node/v14.17.5/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin

Yarn version: 
  1.22.17

Node version: 
  14.17.5

Platform: 
  linux x64

Trace: 
  Error: ENAMETOOLONG: name too long, unlink '/home/username/.cache/yarn/v6/npm-@storybook-react-docgen-typescript-plugin-1.0.2-canary.6.9d540b91e815f8fc2f8829189deb00553559ff63.0-3103532ff494fb7dc3cf835f10740ecf6a26c0f9-integrity/node_modules/@storybook/react-docgen-typescript-plugin'
```

**Comment by FikirYibeltal** [comment]
it seems the problem is encryptfs . I am facing the same issues with ubutntu 20.04

**Comment by FikirYibeltal** [comment, marked as answer]
I have found a work around for the problem. you can move the yarn cache directory from home folder to somewhere in the root directory
using  `yarn config set cache-folder /path `

**Reply by rogix** [reply]
Nice, @FikirYibeltal, thanks! 

Did you get access error? ` An unexpected error occurred: "EACCES: permission denied, mkdir '/path`

**Reply by FikirYibeltal** [reply]
@rogix  yes you have to enable read write access for the directory `sudo chmod 777  /folder`

**Reply by rogix** [reply]
Awesome! It worked for me. Thanks again, @FikirYibeltal!

**Reply by notbear** [reply]
Thanks @FikirYibeltal . I confirm, your workaround worked on Ubuntu 18.04 LTS

**Reply by OmKakatkar** [reply]
Great! @FikirYibeltal . This works on Linux Mint 21.1 Cinnamon

**Reply by aashishsingla567** [reply]
I think it is cacheFolder and not cache-folder now.

**Comment by rafaelsantos-statflo** [comment]
Hey folks, I am having exactly the same issue however, @FikirYibeltal workaround didn't work for me. I am also in a linux mint where the entire partition is encrypted; therefore, moving the yarn cache folder didn't work.

Any timeline on when this should be fixed?

**Comment by birgersp** [comment]
Why are you discussing this here, and not in the [issue itself](https://github.com/storybookjs/storybook/issues/19017)

Upvote the issue please

**Comment by iamabdulhaseeb** [comment]
For me this was happening because I had too many nested ios folders, inside main ios folder I had an ios folder and inside that one there was one more, there were more than dozen ios nested folders, I just deleted the one inside main ios folder which contained other subs, and it started working fine
